core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -323,8 +324,13 @@ private void updateStream() throws IOException {
     /* Create a new stream to read from mPosition. */
     // Calculate block id.
     long blockId = mStatus.getBlockIds().get(Math.toIntExact(mPosition / mBlockSize));
+    BlockInfo blockInfo = mStatus.getBlockInfo(blockId);
+    if (blockInfo == null) {
+      throw new IOException("Block " + blockId + " does not exist");
```

Comment:
can we provide more information about this message, like the path and file id of the file (hopefully they are in `mStatus` already)?